airflow/www/views.py

```diff
@@ -246,10 +266,20 @@ def get_int_arg(value, default=0):
                     DagModel.owners.ilike('%' + arg_search_query + '%')
                 )
 
+            if arg_tags_filter:
+                dags_query = dags_query.filter(DagModel.tags.any(DagTag.name.in_(arg_tags_filter)))
+
             if 'all_dags' not in filter_dag_ids:
                 dags_query = dags_query.filter(DagModel.dag_id.in_(filter_dag_ids))
 
             dags = dags_query.order_by(DagModel.dag_id).offset(start).limit(dags_per_page).all()
+            tags = []
+
+            for name, in session.query(DagTag.name).distinct(DagTag.name).all():
+                value = {'name': name, 'selected': False}
+                if arg_tags_filter and name in arg_tags_filter:
+                    value['selected'] = True
+                tags.append(value)
```

Comment:
Shorter to write:

```python
dagtags = session.query(DagTag.name).distinct(DagTag.name).all()
tags = [{"name": name, "selected": True if arg_tags_filter and name in arg_tags_filter else False}]
```

Reply:
It is shorter but this line it way to long and not very readable IMHO.

Reply:
It's not only shorter but list comprehensions are generally also faster. Does this make it more readable for you? The `True if X else False` can also be replaced by a single `bool()` call.

```python
dagtags = session.query(DagTag.name).distinct(DagTag.name).all()
tags = [
    {"name": name, "selected": bool(arg_tags_filter and name in arg_tags_filter)}
    for name, in dagtags
]
```

Reply:
You're right, fixed.